### PR TITLE
fix: Convert Reddit cluster s2s and p2p to fast

### PR DESCRIPTION
**Pull request by isaac-chung**
## Checklist for adding MMTEB dataset

Resolve https://github.com/embeddings-benchmark/mteb/issues/728

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Clustering/eng/RedditClustering.py [hunk 1]
```diff
@@ -32,3 +33,53 @@ class RedditClustering(AbsTaskClustering):
...
+                        Iryna Gurevych},
+        title     = {TWEAC: Transformer with Extendable QA Agent Classifiers},
+        journal   = {arXiv preprint},
+        volume    = {abs/2104.07081},
+        year      = {2021},
+        url       = {http://arxiv.org/abs/2104.07081},
+        archivePrefix = {arXiv},
+        eprint    = {2104.07081}
+        }""",
+        n_samples={"test": 2048},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/eng/RedditClustering.py:74]: Might be worth increasing this to e.g. 16k (not sure it is needed though)

File: mteb/tasks/Clustering/eng/RedditClustering.py [hunk 2]
```diff
@@ -32,3 +33,53 @@ class RedditClustering(AbsTaskClustering):
...
+        type="Clustering",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["eng-Latn"],
+        main_score="v_measure",
+        date=("2021-01-01", "2021-04-14"),
+        form=["written"],
+        domains=["Web", "Social"],
+        task_subtypes=["Thematic clustering"],
+        license="Not specified",
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/eng/RedditClustering.py:56]:
```suggestion
        license="Not specified", # derived from pushshift
```

File: mteb/tasks/Clustering/eng/RedditClustering.py [hunk 3]
```diff
@@ -32,3 +33,53 @@ class RedditClustering(AbsTaskClustering):
         n_samples={"test": 420464},
         avg_character_length={"test": 64.7},
     )
+
+
+class RedditFastClusteringS2S(AbsTaskClusteringFast):
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/eng/RedditClustering.py:38]: Should we do the P2P dataset as well?

Comment (isaac-chung) [reply mteb/tasks/Clustering/eng/RedditClustering.py:38]: In progress :D you beat me to it

**Comment by isaac-chung** [comment]
When subsampling for the p2p dataset, there are a few labels that only has 1 count. So we get the following error:
```
ValueError: The least populated class in labels column has only 1 member, which is too few. The minimum number of groups for any class cannot be less than 2.
```

Should we exclude them before subsampling? Or force include them somehow?

File: mteb/tasks/Clustering/eng/RedditClustering.py (hunk 1 above)
Comment (KennethEnevoldsen) [reply mteb/tasks/Clustering/eng/RedditClustering.py]: Wonderful @isaac-chung glad to see!

**Comment by KennethEnevoldsen** [comment]
> When subsampling for the p2p dataset, there are a few labels that only has 1 count. 

So what clusters with only one label?

**Comment by isaac-chung** [comment]
> Clusters with only one label?

Yes. You should be able to see this error when you check out the PR and run
```
gh pr checkout 729
mteb -t RedditFastClusteringP2P -m sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2
```

**Comment by KennethEnevoldsen** [comment]
Hmm, that seems weird. This should be there before conversion as well I assume? It might be that the cluster sets does not have the same labels across the sets.

**Comment by isaac-chung** [comment]
The following labels only have 1 sample in the clusters after combining all rows:
```
array(['Bariloche', 'CRISPR', 'RomanianWolves', 'TherosDMs',
       'ValorantBrasil', 'auntienetwork', 'candlemagick', 'graylog',
       'imaginarypenpals', 'visualization'], dtype='<U21')
```

**Comment by KennethEnevoldsen** [comment]
We should probably filter out all labels with less than N. I assume N=1-5 is reasonable?

**Comment by isaac-chung** [comment]
Went with N=1.

**Pull request merged**